src/librustc_resolve/lib.rs

```diff
@@ -2959,6 +2960,16 @@ impl<'a> Resolver<'a> {
                     // `$crate::a::b`
                     module = Some(self.resolve_crate_root(ident.node.ctxt));
                     continue
+                } else if i == 1 && self.session.features.borrow().extern_absolute_paths &&
+                                    path[0].node.name == keywords::CrateRoot.name() &&
```

Comment:
Nit: isn't there some helper function we can use here for this test against `keywords::CrateRoot.name()`?

Reply:
There's no such existing function.
(Also there are too many "crate roots" with different properties now, it's better to be explicit.)